Git-commit: 28176920a0fb56c145d59d3faf164f8d396afd70
From: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Date: Tue, 22 Dec 2020 06:41:01 +0000
Subject: [PATCH] ilp32: skip ARM erratum 1418040 for ilp32 application
Patch-mainline: Queued in openEuler repo, version 5.10.0-1.0.0
References: openEuler-21.03
Git-repo: https://gitee.com/openeuler/kernel.git

hulk inclusion
category: feature
bugzilla: 46790
CVE: NA

--------------------------------

ARM erratum 1418040 only affect AARCH32. ILP32 is not affected.

Signed-off-by: Xiongfeng Wang <wangxiongfeng2@hauwei.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Chen Jun <chenjun102@huawei.com>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 arch/arm64/kernel/process.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/kernel/process.c b/arch/arm64/kernel/process.c
index 1ad01a7887a9..73e3b324c51e 100644
--- a/arch/arm64/kernel/process.c
+++ b/arch/arm64/kernel/process.c
@@ -524,8 +524,8 @@ static void erratum_1418040_thread_switch(struct task_struct *prev,
 	if (!IS_ENABLED(CONFIG_ARM64_ERRATUM_1418040))
 		return;
 
-	prev32 = is_compat_thread(task_thread_info(prev));
-	next32 = is_compat_thread(task_thread_info(next));
+	prev32 = is_a32_compat_thread(task_thread_info(prev));
+	next32 = is_a32_compat_thread(task_thread_info(next));
 
 	if (prev32 == next32 || !this_cpu_has_cap(ARM64_WORKAROUND_1418040))
 		return;
-- 
2.31.1

